lib/dal/src/attribute/prototype/argument.rs

```diff
@@ -234,7 +231,7 @@ impl AttributePrototypeArgument {
         value_source: impl Into<ValueSource>,
     ) -> AttributePrototypeArgumentResult<Self> {
         let argument = Self::new_without_source(ctx, prototype_id, arg_id).await?;
-        Self::set_value_source(ctx, argument.id, value_source).await?;
+        Self::attach_value_source(ctx, argument.id, value_source).await?;
```

Comment:
Minor perf thing: in cases where we know there isn't a value source, it's not worth removing the old edges, so I made an internal attach_value_source that doesn't look for existing edges.